packages/key-management/src/LedgerKeyAgent.ts

```diff
@@ -239,6 +239,9 @@ export class LedgerKeyAgent extends KeyAgentBase {
         this.deviceConnection
       );
       const result = await deviceConnection.signTransaction(ledgerTxData);
+      if (result.txHashHex !== hash.toString()) {
```

Comment:
It would be good to add test for this case.

Reply:
It's a tricky setup in a unit test, for not that much value, as this exception will be covered by integration tests that demonstrate we are able to construct transactions for each property.

Reply:
Added a test in 2ba7be9
Also fixed existing test, it was failing due to using different (mocked) output addresses and producing a different transaction hash